simplecpp.cpp

```diff
@@ -1984,7 +1984,7 @@ namespace simplecpp {
                             if (paren == 0)
                               return tok->next->next;
                             tok = tok->next;
-                            if (parametertokens.front()->next->str() != ")" && parametertokens.size() > args.size())
+                            if (parametertokens.size() > args.size() && parametertokens.front()->next->str() != ")")
```

Comment:
This looks like something we could/should detect. I filed https://trac.cppcheck.net/ticket/12466 about it.